vsintegration/src/FSharp.Editor/IndentationService.fs

```diff
@@ -30,21 +30,33 @@ open Microsoft.FSharp.Compiler.SourceCodeServices
 [<ExportLanguageService(typeof<ISynchronousIndentationService>, FSharpCommonConstants.FSharpLanguageName)>]
 type internal FSharpIndentationService() =
 
-    static member GetDesiredIndentation(sourceText: SourceText, lineNumber: int, tabSize: int): Option<int> =
+    static member GetDesiredIndentation(sourceText: SourceText, lineNumber: int, tabSize: int): Option<int> =        
+        // Match indentation with previous line
+        let rec tryFindPreviousNonEmptyLine l =
+            if l <= 0 then
+                None
+            else
+                let previousLine = sourceText.Lines.[l - 1]
+                if not (String.IsNullOrEmpty(previousLine.ToString())) then                    
+                    Some previousLine
+                else
+                    tryFindPreviousNonEmptyLine (l - 1)
+        // No indentation on the first line of a document
         if lineNumber = 0 then
-            // No indentation on the first line of a document
             None
         else
-            // Match indentation with previous line
-            let previousLine = sourceText.Lines.[lineNumber - 1]
-            let rec loop column spaces =
-                if previousLine.Start + column >= previousLine.End then
-                    spaces
-                else match previousLine.Text.[previousLine.Start + column] with
-                     | ' ' -> loop (column + 1) (spaces + 1)
-                     | '\t' -> loop (column + 1) (((spaces / tabSize) + 1) * tabSize)
-                     | _ -> spaces
-            Some(loop 0 0)
+            match tryFindPreviousNonEmptyLine lineNumber with
+            | None -> Some 0
+            | Some previousLine ->          
+                let rec loop column spaces =
+                    if previousLine.Start + column >= previousLine.End then
+                        spaces
+                    else 
+                        match previousLine.Text.[previousLine.Start + column] with
+                        | ' ' -> loop (column + 1) (spaces + 1)
+                        | '\t' -> loop (column + 1) (((spaces / tabSize) + 1) * tabSize)
```

Comment:
Do we allow \t to appear in an F# source file in the editor?  is it not immediately turned into spaces?